src/monitoring_service/handlers.py

```diff
@@ -184,34 +219,41 @@ def action_monitoring_triggered_event_handler(event: Event, context: Context):
     )
 
     if monitor_request is not None:
-        # FIXME: don't monitor when closer is MR signer
-        contract = context.w3.eth.contract(
-            abi=context.contract_manager.get_contract_abi(
-                CONTRACT_MONITORING_SERVICE,
-            ),
-            address=context.ms_state.blockchain_state.monitor_contract_address,
+        channel = context.db.get_channel(
+            token_network_address=monitor_request.token_network_address,
+            channel_id=monitor_request.channel_identifier,
         )
-        try:
-            tx_hash = contract.functions.monitor(
-                monitor_request.signer,
-                monitor_request.non_closing_signer,
-                monitor_request.balance_hash,
-                monitor_request.nonce,
-                monitor_request.additional_hash,
-                monitor_request.closing_signature,
-                monitor_request.non_closing_signature,
-                monitor_request.reward_amount,
-                monitor_request.token_network_address,
-                monitor_request.reward_proof_signature,
-            ).transact({'gas': 100_000})  # TODO: estimate gas better here
-            log.info(f'Submit MR to SC, got tx_hash {encode_hex(tx_hash)}')
-            assert tx_hash is not None
-            data = context.w3.eth.waitForTransactionReceipt(tx_hash)
-            log.info('MINED', tx=data)
-        except Exception as e:
-            log.error('Sending tx failed', exc_info=True, err=e)
+
+        if channel is not None and channel.closing_tx_hash is None:
```

Comment:
If somebody else provided a more recent BP, we're not allowed to claim. I don't see a check for that case.

Reply:
Should be fixed, pls have another look.